src/GslCore/DnaCreation.fs

```diff
@@ -552,7 +536,29 @@ let expandAssembly
 
                 match ppp.part with
                 | MARKERPART ->
-                    yield expandMarkerPart library dnaSource ppp
+                    // Choose provider
+                    let providers =
+                        markerProviders |>
+                            List.choose (fun provider ->
+                                                    match provider.ScoreJob a.capabilities with
+                                                        | None -> None
+                                                        | Some(score) -> Some (score,provider)

```

Comment:
I'm still confused about how the "score" system is supposed to work.  The score is just a number, and there are no guidelines about how different plugin authors should set the scores relative to each other.  Especially since the marker provider interface has a `IsLegal` method that checks whether the provider can handle the given marker.

Since we're processing a particular marker at this point, it seems like we need to filter the list of available providers based on both the capability score and the ability of each provider to handle this marker, so something like

```
let markerSet = a.pragmas.TryGetOne("markerset") |> Option.defaultValue "default" |> (fun s -> s.ToLower())

let provider =
    markerProviders
    |> List.choose (fun provider ->
        match provider.ScoreJob a.capabilities, provider.IsLegal markerSet with
        | Some(score), true -> Some(score, provider)
        | _ -> None
```
... error handling/etc.